### PR TITLE
Hotfix/3/sc 8931 performance startpage

### PR DESCRIPTION
**Pull request by CeEv**
Todo: 
- [x] Add test

# Description

## Approval for review
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definition of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

**Review by Metauriel — DISMISSED** [review]

File: helpers/cache/Cache.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class Cache {
+	constructor(getFunctions, { updateInvervalSecounds } = {}) {
```
Comment (mwolowyk-cap) [inline helpers/cache/Cache.js]: Typo: 
```suggestion
	constructor(getFunctions, { updateIntervalSecounds } = {}) {
```

File: test/mocha/helpers/cache.test.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
+const { expect } = require('chai');
+
+const Cache = require('../../../helpers/cache/Cache');
+
+
+describe('Cache helper class', () => {
+	it('should return a promise', () => {
+		const callback = () => 'text';
+		const cache = new Cache(callback, { updateInvervalSecounds: 60 });
```
Comment (mwolowyk-cap) [inline test/mocha/helpers/cache.test.js:9]:
```suggestion
		const cache = new Cache(callback, { updateIntervalSecounds: 60 });
```

File: test/mocha/helpers/cache.test.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		const initalData = 'initial';
+		const changedData = 'changed';
+
+		let returnData = initalData;
+		const callback = async () => returnData;
+
+		const cache = new Cache(callback, { updateInvervalSecounds: 60 });
+
+		const result1 = await cache.get();
+		expect(result1, 'that data called direclty.').eql(initalData);
```
Comment (mwolowyk-cap) [inline test/mocha/helpers/cache.test.js:25]:
```suggestion
		expect(result1, 'that data called directly.').eql(initalData);
```

File: test/mocha/helpers/cache.test.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	it('should return a promise', () => {
+		const callback = () => 'text';
+		const cache = new Cache(callback, { updateInvervalSecounds: 60 });
+
+		const resultPromise = cache.get();
+		expect(resultPromise, 'when input is not a promise').to.be.a('promise');
+	});
+
+	it('should return the result first from callback and after it from cache', async () => {
+		const initalData = 'initial';
```
Comment (mwolowyk-cap) [inline test/mocha/helpers/cache.test.js:16]:
```suggestion
		const initialData = 'initial';
```

File: test/mocha/helpers/cache.test.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+		const result1 = await cache.get();
+		expect(result1, 'that data called direclty.').eql(initalData);
+		returnData = changedData;
+
+		const result2 = await cache.get();
+		expect(result2, 'that data retrieved from cache.').eql(initalData);
+	});
+
+	it('should recall ressources if update interval time is expired', async () => {
```
Comment (mwolowyk-cap) [inline test/mocha/helpers/cache.test.js:32]:
```suggestion
	it('should recall resources if update interval time is expired', async () => {
```

**Review by mwolowyk-cap — CHANGES_REQUESTED** [review]

File: helpers/cache/Cache.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+		this.data = null;
+		if (!(typeof callback === 'function')) {
+			throw new Error('First parameter in "Cache" constructor must be a function.');
+		}
+		this.callback = callback;
+	}
+
+	async get(req, res, next) {
+		if (Date.now() > (this.lastUpdated + this.updateInvervalSecounds)) {
+			// please first update time and response old data until request return result
```
Comment (mwolowyk-cap) [inline helpers/cache/Cache.js:14]: do you need please here? :)

**Review by mwolowyk-cap — CHANGES_REQUESTED** [review]

File: test/mocha/helpers/cache.test.js (hunk 2 above)
Comment (CeEv) [reply test/mocha/helpers/cache.test.js:9]: done

File: test/mocha/helpers/cache.test.js (hunk 4 above)
Comment (CeEv) [reply test/mocha/helpers/cache.test.js:16]: done

File: test/mocha/helpers/cache.test.js (hunk 3 above)
Comment (CeEv) [reply test/mocha/helpers/cache.test.js:25]: done

File: test/mocha/helpers/cache.test.js (hunk 5 above)
Comment (CeEv) [reply test/mocha/helpers/cache.test.js:32]: done

File: helpers/cache/Cache.js (hunk 6 above)
Comment (CeEv) [reply helpers/cache/Cache.js:14]: yes :P

**Review by mwolowyk-cap — APPROVED** [review]

File: helpers/cache/Cache.js (hunk 6 above)
Comment (Metauriel) [reply helpers/cache/Cache.js:14]: please update time first. further requests are waiting on the promise.

**Review by Metauriel — APPROVED** [review]

**Pull request merged**